src/connection.ts

```diff
@@ -157,10 +165,21 @@ export class Connection extends EventEmitter<ConnectionEvents> {
 					processedLines++
 				}
 
-				const deserializers = this._getVersionedDeserializers()
-				// attempt to deserialize the response if we can
-				if (deserializers[response.command] && response.data.length) {
-					response.data = await deserializers[response.command](response.data)
+				// Ask what the request was for this response:
+				const sentRequest = this._getRequestForResponse(response)
+				if (sentRequest) {
+					const deserializers = this._getVersionedDeserializers() as {
+						[key: string]: ((input: string[]) => Promise<any>) | undefined
+					}
+					const deserializer = deserializers[sentRequest.command.command]
+					// attempt to deserialize the response if we can
+					if (deserializer && response.data.length) {
```

Comment:
`response.data` can be undefined here. It is _expected_ to exist when we find a deserialiser but if we get an error code back (i.e. an INFO command executed on a non-existing command) it will be undefined.

Probably part of the problem here is that `response.data` is typed as `any`. Perhaps a more suitable type would be `string[] | undefined` although that's technically still incorrect I guess.

Reply:
There is some additional concern on the typings here actually because they assume the request will always return as expected. I.e. a 404 for an info command will have data according to the typings.

Reply:
I made something that is slightly better and (hopefully) more correct, type-wise.